astropy/units/quantity_helper/function_helpers.py

```diff
@@ -556,7 +557,57 @@ def full_impl(shape, fill_value, dtype=None, order='C', *, device=None):
     out_unit = getattr(fill_value, "unit", _UnitFromLikeArg)
     return ret_arr, out_unit, None
 
+@dispatched_function
+def array(object, dtype=None, *, copy=True, order='K', subok=False, ndmin=0):
+    ret_arr = np.array(object, dtype, copy=copy, order=order, subok=subok, ndmin=ndmin)
+    out_unit = getattr(object, "unit", _UnitFromLikeArg)
+    return ret_arr, out_unit, None
+
+if NUMPY_LT_2_0:
+    @dispatched_function
+    def asarray(a, dtype=None, order=None, *, copy=None):
+        return asarray_impl(a, dtype, order, copy=copy)
+else:
+    @dispatched_function
+    def asarray(a, dtype=None, order=None, *, device=None, copy=None):
+        return asarray_impl(a, dtype, order, device=device, copy=copy)
+
+def asarray_impl(a, dtype=None, order=None, *, device=None, copy=None):
```

Comment:
Like my comment for `np.zeros`, etc., I think all these functions could share a common implementation if one passed along `function=np.asarray`, etc.

It is a pity that for these the wrapping with `function_helper` does not work. Really, the `super` call should work - I guess this is a bug on the numpy side?